android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -178,11 +284,51 @@ private fun <T : Comparable<T>> Worker.bind(
    * GlobalScope won't leak the job, because the flow completes when lifecycle completes.
    */
   @OptIn(DelicateCoroutinesApi::class)
-  val job = GlobalScope.launch(RibDispatchers.Unconfined, start = CoroutineStart.UNDISPATCHED) {
+  val job = GlobalScope.launch(
+    coroutineScopeLauncher.coroutineDispatcher,
+    start = coroutineScopeLauncher.coroutineStart
+  ) {
     lifecycle
       .takeWhile { it < lifecycleRange.endInclusive }
-      .onCompletion { onStop() }
-      .collect { onStart(WorkerScopeProvider(lifecycle.asScopeProvider(lifecycleRange))) }
+      .onCompletion {
+        bindAndReportDuration(workerDurationListenerWeakRef, WorkerEvent.STOP) { onStop() }
+      }
+      .collect {
+        bindAndReportDuration(workerDurationListenerWeakRef, WorkerEvent.START) {
+          val workerScopeProvider = WorkerScopeProvider(lifecycle.asScopeProvider(lifecycleRange))
+          onStart(workerScopeProvider)
+        }
+      }
   }
   return WorkerUnbinder(job::cancel)
 }
+
+private inline fun Worker.bindAndReportDuration(
+  workerBinderDurationListeners: WeakReference<WorkerBinderDurationListener>?,
+  event: WorkerEvent,
+  workerBindAction: Worker.() -> Unit
+) {
+  val duration = measureTimeMillis { workerBindAction() }
+  workerBinderDurationListeners.reportDuration(this, event, duration)
```

Comment:
Since this was nullable and weak ref, it needs to be checked before calling the method, right?

Reply:
It seems to be the extension function will trigger only when non nullable, but will update the call per clarity